go-controller/pkg/node/management-port_linux.go

```diff
@@ -311,7 +320,29 @@ func setupManagementPortNFTables(cfg *managementPortConfig) error {
 		),
 	})
 
+	isPodNetworkAdvertised := cfg.isPodNetworkAdvertised.Load()
+
 	if cfg.ipv4 != nil {
+		if isPodNetworkAdvertised {
+			tx.Add(&knftables.Rule{
+				Chain: nftablesMgmtPortChain,
+				Rule: knftables.Concat(
+					"meta nfproto ipv4",
+					"fib saddr type != local",
```

Comment:
for UDNs do we not have different management port chains per udn or what? @tssurya ? Maybe not yet since host -> UDN pod doesnt work?

Reply:
yes, it's done in `getUDNMasqueradeRules`, still relies on iptables, will be handled by @arghosh93 in his work to support UDNs.